cilium/cilium
# bpf: fix mounting of BPF ipcache map in Minikube environments

* Add more logs which indicate when maps are scheduled to be opened, and if they fail to be opened.
* Add new log field for BPF map name.
* Unconditionally open the BPF ipcache map in the datapath. This handles the case where the opening from userspace fails, which occurs in Minikube environments. 

Signed-off by: Ian Vernon <ian@cilium.io>

Fixes: #3519 

```release-note
Unconditionally open the BPF ipcache from the datapath to avoid issues with delayed opening of maps in Minikube
```

**How to test (optional)**:
0. Run step 0 of [this guide](http://cilium.readthedocs.io/en/latest/gettingstarted/minikube/#step-0-install-kubectl-minikube).
1. Run step 1 of [this guide](http://cilium.readthedocs.io/en/latest/gettingstarted/minikube/#step-1-installing-cilium) except use the image `ian-ipcache` for the tag of the Cilium image in the DaemonSet.
2. Once the Cilium pod has entered 'Running' State, exec on the pod as follows:

```
kubectl exec -n kube-system <cilium pod name> -- cilium bpf ipcache list 
```

You should be able to see the output of the contents of the IPcache:
```
$ kubectl exec cilium-gff7m  -n kube-system -- cilium bpf ipcache list
IP ADDRESS           IDENTITY   
10.15.28.238         1          
192.168.99.100       1          
fc00::10ca:1         1          
f00d::a0f:0:0:74ca   38509      
10.15.242.54         38509      
f00d::a0f:0:0:8ad6   1 
```
